### DISCUSSION TITLE
What problems does this library solve?

### DISCUSSION THREAD

**Discussion by david-plugge** [SergeantWeb/sveltekit-plugin-system#4] [General]
I'm still confused why I would need this library. I had a look at the example but I don't understand the main idea. It seems like you can set components to different locations and use them somewhere else. Sounds a little like dependency injection which is a nice thing in general, but since you can not pass any arguments to the component I don't really see a use case.

Please let me know if I just misunderstood the purpose, happy coding :)

**Comment by SergeantWeb** [comment]
It's similar to the use case of WordPress hooks. For example, if you have a main project deployed to multiple locations and need to incorporate various features for each instance, you can implement them with custom plugins.

But you are correct, I overlooked handling component props. Therefore, I have updated the library to support this functionality.
